src/compiler/scala/tools/nsc/transform/LazyVals.scala

```diff
@@ -54,14 +54,35 @@ abstract class LazyVals extends Transform with TypingTransformers with ast.TreeD
     private val lazyVals = perRunCaches.newMap[Symbol, Int]() withDefaultValue 0
 
     import symtab.Flags._
+    private def flattenThickets(stats: List[Tree]): List[Tree] = stats.flatMap(_ match {
```

Comment:
I've just tried this (outside of this PR) here: https://github.com/retronym/scala/compare/6fb41ed...retronym:topic/shrubbery?expand=1

I couldn't find too many spots that were ripe for retrofitting, unfortunately. Can you think of any others? If we can find more clients, the change will be more compelling.